lib/lti.ex

```diff
@@ -52,13 +41,46 @@ defmodule LTI do
   end
 
   defp base_string(%Credentials{url: url}, oauth_params, launch_params) do
+    {normalized_url, query} = parse_url(url)
+    query_params = to_query_params(query)
+
     query =
       oauth_params
-      |> launch_query(launch_params)
+      |> launch_query(launch_params, query_params)
       |> Enum.join("&")
       |> percent_encode()
 
-    "POST&#{percent_encode(url)}&#{query}"
+    "POST&#{percent_encode(normalized_url)}&#{query}"
+  end
+
+  def launch_query(%OAuthData{} = oauth, %LaunchParams{} = launch_params, query_string_params) do
+    parameters = launch_data(oauth, launch_params) ++ query_string_params
+
+    parameters
+    |> Enum.reduce(%{}, fn {key, value}, acc ->
+      Map.put(acc, key, value)
+    end)
+    |> Enum.reduce([], fn {key, value}, acc ->
+      acc ++ ["#{key}=#{percent_encode(value)}"]
+    end)
+  end
+
+  defp parse_url(url) do
+    %URI{scheme: scheme, authority: authority, path: path, query: query} = URI.parse(url)
```

Comment:
Circle ci fails because of length of line. What can be done about this? @fatboypunk 

[R] ↓ Line is too long (max is 80, was 89).